### PR TITLE
[SPARK-45831][CORE][SQL][DSTREAM] Use collection factory instead to create immutable Java collections

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr change to use collection factory instread of `Collections.unmodifiable` to create an immutable Java collection(new collection API introduced after [JEP 269](https://openjdk.org/jeps/269))

### Why are the changes needed?
Make the relevant code look simple and clear.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by LuciferYang** [comment]
There are also some places where `Collections.unmodifiable` is used, such as

https://github.com/apache/spark/blob/bfcd2c47742d69632584f6bb3cf08a8bc3ca8c8c/sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Util.scala#L85-L102

But I don't think the new API can make them look simpler, so I didn't change them in this pr

File: common/utils/src/main/java/org/apache/spark/network/util/JavaUtils.java [hunk 1]
```diff
@@ -202,29 +202,27 @@ private static boolean isSymlink(File file) throws IOException {
...
-    byteSuffixesBuilder.put("m", ByteUnit.MiB);
-    byteSuffixesBuilder.put("mb", ByteUnit.MiB);
-    byteSuffixesBuilder.put("g", ByteUnit.GiB);
-    byteSuffixesBuilder.put("gb", ByteUnit.GiB);
-    byteSuffixesBuilder.put("t", ByteUnit.TiB);
-    byteSuffixesBuilder.put("tb", ByteUnit.TiB);
-    byteSuffixesBuilder.put("p", ByteUnit.PiB);
-    byteSuffixesBuilder.put("pb", ByteUnit.PiB);
-    byteSuffixes = Collections.unmodifiableMap(byteSuffixesBuilder);
+      timeSuffixes = Map.of(
```
Comment (dongjoon-hyun) [inline common/utils/src/main/java/org/apache/spark/network/util/JavaUtils.java:205]: Indentation?

**Review by dongjoon-hyun — COMMENTED** [review]
Are these all occurrences?

**Comment by LuciferYang** [comment]
> There are also some places where `Collections.unmodifiable` is used, such as
> 
> https://github.com/apache/spark/blob/bfcd2c47742d69632584f6bb3cf08a8bc3ca8c8c/sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Util.scala#L85-L102
> 
> But I don't think the new API can make them look simpler, so I didn't change them in this pr



@dongjoon-hyun There are three other similar places here, and I think both `Collections.unmodifiableMap` and `Map.copyOf` can be used, and `Map.copyOf` will trigger a collection copy, so no changes were made in this pr.

**Review by dongjoon-hyun — APPROVED** [review]
Got it. +1, LGTM.

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun ~

**Comment by dongjoon-hyun** [comment]
Merged to master~

**Review by beliefer — COMMENTED** [review]
late LGTM.

**Comment by LuciferYang** [comment]
Thanks @beliefer ~

**Pull request closed without merging**